src/lib/operators/jit_operator_wrapper.cpp

```diff
@@ -52,17 +59,38 @@ std::shared_ptr<const Table> JitOperatorWrapper::_on_execute() {
     context.snapshot_commit_id = transaction_context()->snapshot_commit_id();
   }
 
-  _source()->before_query(in_table, context);
+  _source()->before_query(in_table, _input_parameter_values, context);
   _sink()->before_query(*out_table, context);
 
-  for (auto& jit_operator : _jit_operators) {
+  _prepare_and_specialize_operator_pipeline();
+
+  for (opossum::ChunkID chunk_id{0}; chunk_id < in_table.chunk_count() && context.limit_rows; ++chunk_id) {
+    const auto& in_chunk = *in_table.get_chunk(chunk_id);
+    _source()->before_chunk(in_table, in_chunk, context);
+    _specialized_function_wrapper->execute_func(_source().get(), context);
+    _sink()->after_chunk(*out_table, context);
+  }
+
+  _sink()->after_query(*out_table, context);
+
+  return out_table;
+}
+
+void JitOperatorWrapper::_prepare_and_specialize_operator_pipeline() {
+  // Ensure that a jittable operartor pipeline is only specialized once.
```

Comment:
It took me a while to understand why this might be specialized more than once. Could you explain that this might happen for correlated subqueries? Referring to that explanation in `jit_operator_wrapper.hpp` would be good, too.

Reply:
Also, why would this be called twice; what would we need the mutex for?